rita/src/rita_common/network_monitor/mod.rs

```diff
@@ -180,6 +184,63 @@ impl Default for NetworkMonitor {
     }
 }
 
+#[derive(Default, Serialize, Copy, Clone)]
+pub struct LatencyStats {
+    avg: Option<f32>,
+    std_dev: Option<f32>,
+}
+
+#[derive(Default, Serialize, Copy, Clone)]
+pub struct PacketLossStats {
+    avg: Option<f32>,
+    five_min_avg: f32,
+}
+
+pub struct GetStats {
+    pub routes: Vec<BabelRoute>,
+}
+
+#[derive(Serialize, Default, Copy, Clone)]
+pub struct IfaceStats {
+    latency: LatencyStats,
+    packet_loss: PacketLossStats,
+}
+
+impl Message for GetStats {
+    type Result = Result<Stats, Error>;
+}
+
+pub type Stats = HashMap<String, IfaceStats>;
+
+impl Handler<GetStats> for NetworkMonitor {
+    type Result = Result<Stats, Error>;
+
+    fn handle(&mut self, msg: GetStats, _ctx: &mut Context<Self>) -> Self::Result {
+        let mut stats = Stats::new();
+
+        for route in msg.routes.iter() {
```

Comment:
Somehow you want to combine these two lists without unwrap (because unwrap crashes, what if someone views neighbors while and we have a latency stats entry for an interface but not a packet loss stats entry? In theory this should never happen because of the way actix works, but the best way to write code that works is to assume someone's going to go around breaking things like that. 

for (iface, latency_stats) self.latency_history.iter() {
if let Some(packet_stats) = self.packet_loss_history.get(&iface) {
stats.insert(...
}
else {
error!("Found entry in one that's not in the other ") // do this error message better but you get the idea
}
}

this way we have no unwraps, and a simpler request from the other side. 

https://users.rust-lang.org/t/how-to-iterate-through-two-arrays-at-once/783/3

This could be used to zip the iterators, or you could iter over one and if match the other, or a match statement where each one uses .iter().next() (to get the next entry). But this is syntactically complicated although pretty if you can do it. Match statements are really awesome. They take complicated if/else trees and turn them into giant pattern matching self checking systems. Kinda like switch statements but more awesome.